File: src/main/java/com/mallang/auth/presentation/AuthContext.java
package com.mallang.auth.presentation;

import com.mallang.auth.exception.IncorrectUseAuthAtException;
import org.springframework.stereotype.Component;
import org.springframework.web.context.annotation.RequestScope;

@RequestScope
@Component
public class AuthContext {

    private Long memberId;

    public boolean unAuthenticated() {
        return memberId == null;
    }

    public Long getMemberId() {
        if (unAuthenticated()) {
            throw new IncorrectUseAuthAtException();
        }
        return memberId;
    }

    public void setMemberId(Long memberId) {
        this.memberId = memberId;
    }
}

File: src/main/kotlin/com/mallang/auth/presentation/support/AuthContext.kt
package com.mallang.auth.presentation.support

import com.mallang.auth.exception.IncorrectUseAuthAtException
import org.springframework.stereotype.Component
import org.springframework.web.context.annotation.RequestScope


@RequestScope
@Component
class AuthContext(
        private var memberId: Long? = null
) {

    fun unAuthenticated(): Boolean {
        return memberId == null
    }

    fun getMemberId(): Long {
        return memberId ?: throw IncorrectUseAuthAtException()
    }

    fun setMemberId(memberId: Long) {
        this.memberId = memberId
    }
}
